console/v1/types_console_link.go

```diff
@@ -47,7 +47,14 @@ type ApplicationMenuSpec struct {
 // NamespaceDashboardSpec is a specification of namespaces in which the dashboard link should appear.
 type NamespaceDashboardSpec struct {
 	// namespaces is an array of namespace names in which the dashboard link should appear.
-	Namespaces []string `json:"namespaces"`
+	Namespaces []string `json:"namespaces,omitempty"`
+	// namespaceSelector is used to filter the Namespaces that should contain dashboard link.
+	// If the namespace labels match, dasboard link will be shown for the namespaces.
+	//
+	// If unset, dashboard link should appear in all the namespaces from the namespaces array.
+	//
+	// +optional
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
```

Comment:
Was thinking about the name and think that the `namespaceSelector` is suitable, since we use it in other place in our API.